src/Orleans.Runtime/Streams/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -190,15 +133,18 @@ public LeaseBasedQueueBalancer(IServiceProvider serviceProvider, ISiloStatusOrac
         }
 
         /// <inheritdoc/>
-        public override Task Initialize(string strProviderName, IStreamQueueMapper queueMapper, TimeSpan siloMaturityPeriod, IProviderConfiguration providerConfig)
+        public override Task Initialize(string strProviderName, IStreamQueueMapper queueMapper, TimeSpan siloMaturityPeriod)
         {
             if (queueMapper == null)
             {
                 throw new ArgumentNullException("queueMapper");
             }
-            var balancerConfig = new LeaseBasedQueueBalancerConfig(providerConfig);
-            this.leaseProvider = this.serviceProvider.GetRequiredService(balancerConfig.LeaseProviderType) as ILeaseProvider;
-            this.leaseLength = balancerConfig.LeaseLength;
+            var options = this.serviceProvider.GetServiceByName<LeaseBasedQueueBalancerOptions>(strProviderName)
```

Comment:
we should have providerBuilder and provider lifecycle. :) 

There's too many components in streams. The complexity can benefits from lifecycle and builder pattern, in both perspective of configuring and start and shutdown sequence. 

Some of those are managed by Silo(Client)ProviderRuntime, some are managed by the PersistentStreamProvider. Those unnecessary coupling partially result in bad consequence, such as streaming infra has to live in core. 

Just throwing ideas out there, not necessary the scope of this PR.

Reply:
I'm not convinced of the lifecycle aspect, but it's definitely worth consideration.  In the description I've started a list of further refactoring we may want to do in the future.  Please feel free to edit the description to add more suggestions.